src/data-table/__tests__/data-table-columns.e2e.js

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright (c) 2018-2019 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+/* eslint-env node */
+/* eslint-disable flowtype/require-valid-file-annotation */
+
+const {mount, analyzeAccessibility} = require('../../../e2e/helpers');
+
+function getHeaderCellAtIndex(page, index) {
+  return page.$(
+    // plus one to convert to one indexed item
+    `div[data-baseweb="data-table"] > div > div:nth-child(${index + 1})`,
+  );
+}
+
+function getCellAtIndex(page, index) {
+  // plus two to convert to one indexed item and skips header row
+  return page.$(`div[data-baseweb="data-table"] > div:nth-child(${index + 2})`);
+}
+
+function getCellsAtColumnIndex(page, index) {
+  // hard-coded related column count for now. could be calcuated by number of children if scenario changes.
+  const COLUMN_COUNT = 4;
+  const indices = [];
+  for (let i = 0; i < COLUMN_COUNT; i++) {
+    indices.push(i * COLUMN_COUNT + index);
+  }
+  return Promise.all(indices.map(i => getCellAtIndex(page, i)));
+}
+
+function getTextContentFromElements(page, elements) {
+  return Promise.all(
+    elements.map(element => {
+      return page.evaluate(e => e.textContent, element);
+    }),
+  );
+}
+
+async function getCellContentsAtColumnIndex(page, index) {
+  const elements = await getCellsAtColumnIndex(page, index);
+  return getTextContentFromElements(page, elements.filter(Boolean));
+}
+
+async function sortColumnAtIndex(page, index) {
+  const headerCell = await getHeaderCellAtIndex(page, index);
+  const sortButton = await headerCell.$('div[role="button"]');
+  return sortButton.click();
+}
+
+async function openFilterAtIndex(page, index) {
+  const headerCell = await getHeaderCellAtIndex(page, index);
+  await headerCell.hover();
+  const hoveredHeaderCell = await getHeaderCellAtIndex(page, index);
+  const filterButton = await hoveredHeaderCell.$('button');
+  await filterButton.click();
+  return page.$('div[data-baseweb="popover"]');
+}
+
+function matchArrayElements(a, b) {
+  if (a.length !== b.length) return false;
+  for (let i = 0; i < a.length; i++) {
+    if (a[i] !== b[i]) return false;
+  }
+  return true;
+}
+
+describe('data table columns', () => {
+  it('passes basic a11y tests', async () => {
+    await mount(page, 'data-table-columns');
+    const accessibilityReport = await analyzeAccessibility(page);
+    expect(accessibilityReport).toHaveNoAccessibilityIssues();
+  });
+
+  it('renders expected number of cells', async () => {
+    await mount(page, 'data-table-columns');
+    await page.waitFor('div[data-baseweb="data-table"]');
```

Comment:
nit: may make sense to move this to a const to the top of the file, as it used a few times